pep-0702.rst

```diff
@@ -0,0 +1,278 @@
+PEP: 702
+Title: Typing support for deprecations
+Author: Jelle Zijlstra <jelle.zijlstra@gmail.com>
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Content-Type: text/x-rst
+Created: 30-Dec-2022
+Python-Version: 3.12
+
+
+Abstract
+========
+
+This PEP adds an ``@typing.deprecated()`` decorator that marks a class or function
+as deprecated, enabling static checkers to warn when it is used.
+
+Motivation
+==========
+
+As software evolves, new functionality is added and old functionality becomes
+obsolete. Library developers want to work towards removing obsolete code while
+giving their users time to migrate to new APIs. Python provides a mechanism for
+achieving these goals: the :exc:`DeprecationWarning` warning class, which is
+used to show warnings when deprecated functionality is used. This mechanism is
+widely used: as of the writing of this PEP, the CPython main branch contains
+about 150 distinct deprecations. Many third-party libraries also use
+:exc:`DeprecationWarning` to mark deprecations.
```

Comment:
Is it easy to get some numbers for e.g. the 1000 most popular PyPI packages?

Reply:
https://dev.to/hugovk/how-to-search-5000-python-projects-31gk specifically, I'm currently at 472/5000 downloaded and will report back when it is finished :)

Reply:
Added some numbers (at least 3.5k deprecations in the top 5k packages).